### ISSUE TITLE
This send_object_list and recv_object_list method has performance problems.

### ISSUE THREAD

**Issue by fujl** [pytorch/pytorch#134390]
### 🐛 Describe the bug

send_object_list
============pickle dumps used time 0.00702357292175293 logits_len=64
============pickle dumps used time 0.012560129165649414 logits_len=64
============pickle dumps used time 0.017546653747558594 logits_len=64
recv_object_list
pickle loads rank=1 used time 0.0048525333404541016
pickle loads rank=2 used time 0.004410505294799805
pickle loads rank=3 used time 0.004876375198364258

### Versions

pytorch 2.4.0
Serialization and deserialization are slow.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

**Comment by yifuwang** [comment]
When using nccl, `send_object_list` pickles the message, copies it to device, then sends it via nccl. `recv_object_list` receives the message via nccl, copies it to host, them unpickles the message. That one device sync on each end.

If the object contains any GPU tensors, the situation becomes worse: the sender needs to copy the data from device to host during pickling, and the receiver needs to copy the data from host to device during unpickling. This adds more device syncs.

Seems like you are just sending logits? Can you just use send/recv as opposed to their _object_list variant? If you give us a code snippet we can better help you.

**Comment by wconstab** [comment]
closing as answered, please reopen if you disagree
